ADAL/src/cache/ios/ADKeychainTokenCache.m

```diff
@@ -435,26 +435,18 @@ - (BOOL)removeAllForUserId:(NSString * __nonnull)userId
     return YES;
 }
 
-- (BOOL)removeAllForUserId:(NSString *)userId error:(ADAuthenticationError *__autoreleasing  _Nullable *)error
+- (BOOL)wipeAllItemsForUserId:(NSString *)userId error:(ADAuthenticationError *__autoreleasing  _Nullable *)error
 {
     AD_LOG_WARN(nil, @"Removing all items for user.");
     AD_LOG_WARN_PII(nil, @"userId <%@>", userId);
 
-    NSArray *items = [self allItems:nil];
-    if (!items)
-    {
-        return NO;
-    }
-    
-    for (ADTokenCacheItem *item in items)
+    @synchronized(self)
```

Comment:
Silly question, why?

Reply:
Not silly Q, silly implementation. No need to have synchronized here.